src/templates/nulldata.js

```diff
@@ -14,15 +14,23 @@ function check (script) {
 check.toJSON = function () { return 'null data output' }
 
 function encode (data) {
-  typeforce(types.Buffer, data)
+  typeforce(typeforce.oneOf(types.Buffer, types.Array), data)
 
-  return bscript.compile([OPS.OP_RETURN, data])
+  if (!types.Array(data)) {
+    data = [ data ]
+  }
+
+  return bscript.compile([OPS.OP_RETURN].concat(data))
 }
 
 function decode (buffer) {
   typeforce(check, buffer)
 
-  return buffer.slice(2)
+  var chunks = bscript.decompile(buffer)
+
+  chunks.shift()
```

Comment:
@dabura667 it is an unfortunate optimization that has lead to way too much confusion.